airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,184 @@
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ssh_hook = ssh_hook
+        self.ssh_conn_id = ssh_conn_id
+        self.remote_host = remote_host
+        self.local_path = local_path
+        self.remote_path = remote_path
+        self.regexp_mask = regexp_mask
+        self.operation = operation
+        self.confirm = confirm
+        self.create_intermediate_dirs = create_intermediate_dirs
+        if not (self.operation.lower() == SFTPOperation.GET or self.operation.lower() == SFTPOperation.PUT):
+            raise TypeError(
+                f"""Unsupported operation value {self.operation},
+                expected {SFTPOperation.GET} or {SFTPOperation.PUT}"""
+            )
+        if not (
+            (isinstance(self.local_path, str) and isinstance(self.remote_path, str))
+            or (isinstance(self.local_path, list) and isinstance(self.remote_path, str))
+            or (isinstance(self.remote_path, list) and isinstance(self.local_path, str))
+        ):
+            raise TypeError(
+                """Unsupported path argument value local_path and remote_path
+                Possible options: \n local_path is str and remote_path is str\n
+                local_path is list and remote_path is str\n
+                local_path is str and remote_path is list"""
+            )
+
+    def execute(self, context: Any) -> str:
+        file_msg = None
+        try:
+            _check_conn(self)
+
+            with self.ssh_hook.get_conn() as ssh_client:
+                sftp_client = ssh_client.open_sftp()
+                if self.operation.lower() == SFTPOperation.PUT:
+                    if isinstance(self.local_path, str):
+                        files_list = self._search_files(os.listdir(self.local_path))
+                        for file in files_list:
+                            local_file = os.path.basename(file)
+                            file_msg = file
+                            self._transfer(sftp_client, self.local_path, local_file, self.remote_path)
+                    if isinstance(self.local_path, list) and isinstance(self.remote_path, str):
+                        for file in self.local_path:
+                            local_file = os.path.basename(file)
+                            file_msg = file
+                            self._transfer(sftp_client, os.path.dirname(file), local_file, self.remote_path)
+                elif self.operation.lower() == SFTPOperation.GET:
+                    if isinstance(self.remote_path, str):
+                        files_list = self._search_files(sftp_client.listdir(self.remote_path))
+                        for file in files_list:
+                            remote_file = os.path.basename(file)
+                            file_msg = file
+                            self._transfer(sftp_client, self.local_path, remote_file, self.remote_path)
+                    if isinstance(self.remote_path, list) and isinstance(self.local_path, str):
```

Comment:
if there directory is not exist and create_intermediate_dirs is false, then there will be an error at the time of file transfer, is it worth overloading the method and doing explicit checks for everything?

Reply:
When you get a good error message, that's fine, 

But  I am afraid that when you try to write many files to a file that already exists, it will not error out but it will save all the files one-by-one as the same target file - and will not produce an error. Am I right about it?

Reply:
i checked this point, if the file already exists then it will be overwritten. Since this can be viewed as a feature, I will add an additional parameter `force` so that with such, either give an error or overwrite

Reply:
done ;)